### PR TITLE
gh-114329: Fix PyList_GetItemRef() limited C API definition

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-114329

### CODE REVIEW

File: Modules/_testlimitedcapi/list.c [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// Need limited C API version 3.13 for PyList_GetItemRef()
+#if !defined(Py_GIL_DISABLED) && !defined(Py_LIMITED_API )
```
Comment (colesbury) [inline Modules/_testlimitedcapi/list.c]: You might need an `#include "pyconfig.h"` to get `Py_GIL_DISABLED`

Comment (vstinner) [reply Modules/_testlimitedcapi/list.c]: oops, nicely spotted!

File: Modules/_testlimitedcapi/list.c [hunk 2]
```diff
@@ -1,3 +1,9 @@
+// Need limited C API version 3.13 for PyList_GetItemRef()
+#include "pyconfig.h"   // Py_GIL_DISABLED
+#if !defined(Py_GIL_DISABLED) && !defined(Py_LIMITED_API )
```
Comment (colesbury) [inline Modules/_testlimitedcapi/list.c]: There's an extra space in `defined(Py_LIMITED_API )`

**Review by colesbury — APPROVED** [review]
Thanks! Possible typo below:

File: Modules/_testlimitedcapi/list.c (hunk 2 above)
Comment (vstinner) [reply Modules/_testlimitedcapi/list.c]: Oh. I removed the space in multiple C and header files.

**Pull request merged**